Implement a ClientHttpRequestFactory which uses AsyncHttpClient

bino013 opened the pull request:
My team have implemented a ClientHttpRequestFactory which uses  AsyncHttpClient for async http calls.

spring-builds committed:
Release version 5.0.5.RELEASE

snicoll committed:
Initiate 5.0.x branch

sdeleuze committed:
Fix Dokka reference to Spring Framework's Javadoc

This commit specifies a local packageListUrl and defines that dokka task
must be executed after the api task in order to be able to build KDoc
during the release process when the Spring Framework's Javadoc is not
published yet.

Issue: SPR-16687

sdeleuze committed:
Cleanup settings.gradle pluginManagement configuration

rstoyanchev committed:
TestDispatcherServlet unwraps to find mock request

Issue: SPR-16695

hojak99 committed:
Fix Java 9 link in the reference documentation

sdeleuze committed:
Improve Kotlin + bean validation documentation

Issue: SPR-16701

sdeleuze committed:
Document why "charset=UTF-8" is specified for JSON

Issue: SPR-14715

jhoeller committed:
Remove bogus DataSource test from JpaTransactionManagerTests

(cherry picked from commit ff53d78)

jhoeller committed:
Unwind _TestTypes to top-level public test classes in AOP test suite

(cherry picked from commit cdaa247)

jhoeller committed:
AnnotationUtils.getAnnotation non-null check for synthesizeAnnotation

Issue: SPR-16708

(cherry picked from commit da80502)

jhoeller committed:
Polishing

jhoeller committed:
Upgrade to Netty 4.1.23 and TestNG 6.14.3

sdeleuze committed:
Add default ctor to Reactive UrlBasedCorsConfigurationSource

Issue: SPR-16712

jhoeller committed:
Proper exception for controller method return types that do not work with MvcUriComponentsBuilder (e.g. final classes)

Includes direct use of ControllerMethodInvocationInterceptor for return type Object, avoiding the attempt to generate an Object subclass.

Issue: SPR-16710

(cherry picked from commit f28a5d0)

jhoeller committed:
Upgrade to Tomcat 8.5.30

rstoyanchev committed:
Avoid inifinite recursion in UndertowServerHttpResponse

Undertow does not provide a way to check if we can write so with the
current implementation of isWritePossible, deep recursion can occur
when writing slows down. We now use a flag to keep track of write
ChannelListener callbacks.

This commit also addresses a related issue in
AbstractListenerWriteProcessor that went undected since #3c2d186
where after a large (single) buffer that is not written fully, the
completion signal is processed before the all data is written.

Issue: SPR-16702

rstoyanchev committed:
Remove write pausing in Undertow response

Using the simple example shown in the ticket but switching from
Mono<String> to Flux<String> (and 5,000,000 onNext calls) shows that
constant pausing causes significant overhead and is not worth the
trouble vs ignoring the onWritePossible in REQUESTED state.

Issue: SPR-16702

igor-suhorukov committed:
Throw exception from user code in SpringFailOnTimeout even if a timeout occurs

Issue: SPR-16717

sbrannen committed:
Expand scope of SpringFailOnTimeoutTests

Issue: SPR-16716

jhoeller committed:
Revised reference example for linkable controller method signature

Issue: SPR-16710

(cherry picked from commit 7ee6130)

jhoeller committed:
Fine-tuned JCA MessageEndpoint exception logging and propagation

Issue: SPR-16717

(cherry picked from commit 8e1ecec)

jhoeller committed:
Consistent getTypeForFactoryMethod result for parameterized method

Issue: SPR-16720

(cherry picked from commit 6184c4e)

bclozel committed:
Switch to Reactor Bismusth SNAPSHOTs

jhoeller committed:
Cache-safety check for sibling loaders resolving the same classes

Issue: SPR-16714

nkjackzhang committed:
Fix typo in javadoc

Closes gh-1791

sbrannen committed:
Suppress warning in SpringFailOnTimeoutTests

jhoeller committed:
AspectJExpressionPointcut consistently resolves superinterface methods

Includes efficient check for same ClassLoader in ClassUtils.isVisible, efficient MethodMatchers check for IntroductionAwareMethodMatcher, and supertype method resolution in MethodMapTransactionAttributeSource.

Issue: SPR-16723

(cherry picked from commit b95e05d)

jhoeller committed:
Polishing

(cherry picked from commit de4ff4b)

jhoeller committed:
Local XMLUnit dependency declarations with consistent version 2.5.1

Includes upgrade to Undertow 1.4.24.

(cherry picked from commit 0754833)

rstoyanchev committed:
Restore handling of 0 bytes read

Issue: SPR-16728

rstoyanchev committed:
Selector header name is exposed for configuration

Issue: SPR-16732

jhoeller committed:
Restore original MethodMapTransactionAttributeSource matching rules

Issue: SPR-16733

(cherry picked from commit c5b524d)

jhoeller committed:
OperatorMatches flags misguided evaluation attempts as FLAWED_PATTERN

Issue: SPR-16731

(cherry picked from commit d4a55a2)

sdeleuze committed:
Make ResponseSpec.expectBody Kotlin extension usable

Prior to this commit, due to KT-5464 type inference issue there was not
proper way to provide body expectations with WebTestClient. This commit
provides a workaround by updating the existing Kotlin extension to
return a Kotlin compatible API.

Issue: SPR-15692

jhoeller committed:
Workaround for generic parameter types on inner class constructors

Issue: SPR-16734

jhoeller committed:
Avoid repeated superclass introspection in findAnnotation(Method,...)

Issue: SPR-16730

(cherry picked from commit d78e27f)

jhoeller committed:
Correctly delegate to OrderUtils.getPriority for DecoratingProxy

Issue: SPR-16739

(cherry picked from commit 2f4010e)

rstoyanchev committed:
Avoid creating Exception instance if not needed

Issue: SPR-16726

rstoyanchev committed:
Polish (minor) in AbstractMessageReaderArgumentResolver

nkjackzhang committed:
Fix a typo in @Nullable Javadoc

kiftio committed:
Polish WebFlux reference documentation

rstoyanchev committed:
Polish tests to use WebClient retrieve()

rstoyanchev committed:
Use StringDecoder to split SSE stream

ServerSentEventHttpMessageReader had logic to split on new lines
and buffer until an empty new line (start of a new event). To account
for random data chunking, it later re-assembled the lines for each
event and split again on new lines. However bufferUntil was still
unreliable a chunk may contain nothing but a newline, which doesn't
necessarily mean an empty newline in the overall SSE stream.

This commit simplifies the above by delegating the splitting of the
stream along newlines to StringDecoder.

Issue: SPR-16744

rstoyanchev committed:
In 5.0.x we don't have Flux/Mono error with Supplier

Issue: SPR-16726

snicoll commented:
@bino013 please review this PR, there are a lot of unrelated commits there. Please rebase on master and only push your changes here (you can force push on your branch).

bclozel committed:
Upgrade to Netty 4.1.24.Final

dimitrisli committed:
Fix typo

See gh-1803

sdeleuze committed:
Enable KotlinScriptTemplateTests after KT-18833 fix

sdeleuze committed:
Upgrade Kotlin to 1.2.40

bclozel committed:
Disable HTTP Range support for InputStreamResource

Prior to this commit, the `AbstractMessageConverterMethodProcessor`
would fail to convert `InputStreamResource` to `ResourceRegion` as
expected, since the content length cannot be read without consuming the
stream. This is enforced by the `HttpRange` class.

Now the method processor would still try to output HTTP range response
headers to provide range support information. This step is using the
resource content length and reads the input stream, leading to
exceptions such as "IllegalStateException: InputStream has already been
read".

This commit improves the return type detection and excludes early
`InputStreamResource` return types. With those types, HTTP range support
is now completely disabled.

Issue: SPR-16754
(cherry picked from commit e9a8a5065b)

rstoyanchev committed:
Validate contextPath in RedirectView

Issue: SPR-16752

rstoyanchev committed:
Add more detail to ISE in ServerEndpointExporter

Issue: SPR-16655

hasheniuk committed:
Fix typo

Closes gh-1804

hasheniuk committed:
Improve performance of StringUtils#trimWhitespace
Issue: SPR-16766

(cherry picked from commit 6545cab)

nkjackzhang committed:
Fix typos

Closes gh-1806

rstoyanchev committed:
Better assertion message in MockPart

Issue: SPR-16767

nkjackzhang committed:
Fix typos in Spring MVC refdoc

1. Consistent with "xml code" examples.
2. "xml()" is a static method and will use default builder config,
   so use createXmlMapper(true) instead.
3. Fix mvc namespace tag typo.

sdeleuze committed:
Reuse PartBodyStreamStorageFactory in SynchronossPartGenerator

Issue: SPR-16727

izeye committed:
Polish

nkjackzhang committed:
Fix broken anchor link in WebFlux refdoc

sdeleuze committed:
Fine tune WebFlux server logging verbosity

With this commit, WebFlux server uses warning instead of error log level
for request handling, and also just print the message instead of the
stacktrace which is mostly meaningless in reactive world.

Complementary to this change, Reactor Netty removed additional logging
as part of https://github.com/reactor/reactor-netty/issues/339.

Issue: SPR-16688

jhoeller committed:
AopUtils.getMostSpecificMethod exposes dynamic proxy class methods

Includes efficient canApply check for IntroductionAwareMethodMatcher.

Issue: SPR-16757

(cherry picked from commit aa11721)

jhoeller committed:
Consistent target method resolution for event and caching expressions

Issue: SPR-16779

(cherry picked from commit eaff2c2)

jhoeller committed:
Support for non-standard HTTP status in reactive ClientHttpResponse

Issue: SPR-16748

(cherry picked from commit a683472)

jhoeller committed:
Fine-tuned assertions and related polishing in WebFlux builders

(cherry picked from commit 9bff5b4)

Nickolas-Evans committed:
Fix typo

Closes gh-1813

snicoll committed:
Keep YAML entries that haven an empty array value

Prior to this commit, a YAML entry that define an empty array value was
lost. This commit makes sure to flag it with an empty String, which
corresponds as an empty comma separated list of entries in the
properties format.

Issue: SPR-16769

sdeleuze committed:
Upgrade to Kotlin 1.2.41

Fixes KT-23973 critical regression

rstoyanchev committed:
Consistent handling of URISyntaxException

Issue: SPR-16778

rstoyanchev committed:
ReactorNettyWebSocketSession implements close properly

Issue: SPR-16774

rstoyanchev committed:
Polish WebSocketIntegrationTests

jhoeller committed:
SimpleClientHttpResponse catches any Exception on close

Issue: SPR-16773

(cherry picked from commit 21fad8e)

jhoeller committed:
Nullable HttpMethod parameter only on internal doExecute delegate

Issue: SPR-15540

(cherry picked from commit f8c2d7a)

sdeleuze committed:
Support decoding Mono in Jaxb2XmlDecoder

Issue: SPR-16759

jhoeller committed:
Introspect originating bean definition as configuration class candidate

Issue: SPR-16756

(cherry picked from commit c8b6233)

jhoeller committed:
Expose FactoryBean's raw object on retrieval during post-processing

Issue: SPR-16783

(cherry picked from commit 9281f82)

jhoeller committed:
Upgrade to RxJava 2.1.13, Hibernate ORM 5.2.17, AspectJ 1.9.1

jhoeller committed:
Consistent SpelEvaluationException messages in findAccessorForMethod

Issue: SPR-16762

jhoeller committed:
Lenient fallback to plain getBundle call without Control handle

Issue: SPR-16776
